include/Matrix.cpp

```diff
@@ -0,0 +1 @@
+#include <Matrix.hpp>
```

Comment:
There's no need for `include/Matrix.cpp` file.

Reply:
This file has been deleted

Reply:
@ashok-arora I made an edit to CMakeLists.txt. I'm not too familiar with CMake but I assumed that it was because the include directory was not being linked to within the CMakeLists.txt, so I linked it accordingly. Please let me know whether or not this was a sufficient change, again I am sorry if it's not, I'm not really familiar with how CMake works

Reply:
I tried running with the change: `link_directories(${Matrix_SOURCE_DIR}/include)` but it did not compile. 

Here are the changes that you will need to make along with an explanation: 

1. We will be using CMake's object libraries, you can read more about them [here](https://cmake.org/cmake/help/latest/command/add_library.html#object-libraries).
2. The idea is that the repository has about 40 files that depend on `matrix_basic.cpp` and `matrix_operations.cpp`. 
 One option would be to build them every time for 40 files **or** we can build them beforehand and use as required. 
 Since the second option saves time during compilation, I will go with that.  
3. An object-library can be created in the `CMakeLists.txt` using `add_library(MAT OBJECT ${Matrix_SOURCE_DIR}/include/matrix_basic.cpp ${Matrix_SOURCE_DIR}/include/matrix_operations.cpp)`
4. Then, add ` $<TARGET_OBJECTS:MAT>` to each executable in the respective `examples/CMakeLists.txt` and `benchmarks/CMakeLists.txt` that uses those two files. So something like this: 
```cmake
add_executable(abs abs.cpp $<TARGET_OBJECTS:MAT>)
add_executable(addition addition.cpp $<TARGET_OBJECTS:MAT>)
```
Let me know if you need any further help or explanation on this.

Reply:
Also, I see that your fork is behind `main` by a few commits so it would be better if you could rebase it to avoid merge conflicts.

Reply:
I attempted to resolve the issue with what you've given me. I'm still not able to compile and test so if you could let me know if it works

Reply:
Firstly, could you rebase your fork with the `main` branch since there are merge conflicts. 

After that, since you have taken care of the main issue, I will add a commit to fix the CMake problems.